stingray/largememory.py

```diff
@@ -0,0 +1,558 @@
+                                   data=ev.ncounts,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    meta_data_group.create_dataset(name='mjdref',
+                                   data=ev.mjdref,
+                                   compressor=compressor,
+                                   overwrite=True)
+
+    if ev.energy:
+        main_data_group.create_dataset(name='energy',
+                                       data=ev.energy,
+                                       compressor=compressor,
+                                       overwrite=True,
+                                       chunks=(8388608, ))
+
+    # FIXME: GTI's are not consistently saved
+    if ev.gti:
+        main_data_group.create_dataset(name='gti', data=ev.gti, overwrite=True)
+
+    if ev.pi:
+        main_data_group.create_dataset(name='pi_channel',
+                                       data=ev.pi,
+                                       compressor=compressor,
+                                       overwrite=True)
+
+
+def _combineSpectra(final_spectra):
+    """
+    Create a final spectra that is the mean of all spectra.
+    Parameters
+    ----------
+    final_spectra: :class:`stingray.AveragedCrossspectrum/AveragedPowerspectrum' object
+        Summed spectra of all spectra
+    Returns
+    -------
+    object
+        Final resulting spectra.
+    """
+    final_spectra.freq /= final_spectra.m
+    final_spectra.power /= final_spectra.m
+    final_spectra.unnorm_power /= final_spectra.m
+    # REVIEW: final_spectra.power_err /= final_spectra.m
+
+
+    # Don't look for pds1 and pds2 if it's a power spectrum.
+    # Since Powerspectrum is also a Crossspectrum, we need to specify this
+
+    if isinstance(final_spectra, stingray.AveragedCrossspectrum) and not \
+            isinstance(final_spectra, stingray.AveragedPowerspectrum):
+        final_spectra.pds1.power /= final_spectra.m
+        final_spectra.pds2.power /= final_spectra.m
+
+    return final_spectra
+
+
+def _addSpectra(final_spectra, curr_spec, flag):
+    """
+    Add various Spectra(AveragedCrossspectrum/AveragedPowerspectrum) for combination.
+
+    Parameters
+    ----------
+    final_spectra: object
+        Final Combined AveragedCrossspectrum or AveragedPowerspectrum
+    curr_spec: object
+        AveragedCrossspectrum/AveragedPowerspectrum to be combined
+    flag: bool
+        Indicator variable
+
+    Returns
+    -------
+    object
+        Combined AveragedCrossspectrum/AveragedPowerspectrum
+    """
+    if flag:
+        final_spectra = curr_spec
+        final_spectra.freq = final_spectra.freq.astype('float128')
+        final_spectra.power = final_spectra.power.astype('complex256')
+        final_spectra.unnorm_power = final_spectra.unnorm_power.astype(
+            'complex256')
+
+        return final_spectra
+
+    np.multiply(np.add(final_spectra.freq, curr_spec.freq),
```

Comment:
In order for this to work, all frequency arrays have to be _identical_. Please test that they are, and maintain the `.freq` attribute of the current spectrum

Reply:
Identical as in shape right?

Reply:
No, identical in value. Each FFT has to be over exactly the same frequency range, in each chunk

Reply:
This only applies to frequency and not to any other attribute, right?

Reply:
Yes, just the frequency